### PR TITLE
Config/task updates for creating Q4's IRS instances

### PR DESCRIPTION
**Pull request by CyclopsRex**
@ioanaroceanu2000 please can you review https://github.com/Voltz-Protocol/voltz-core/commit/af8aec11b608ee09e47264fa5cb657ebad80e457 carefully because it changes some of the work you did for the Periphery and I'd like you to check I haven't made things worse rather than better. :)

### CODE REVIEW

**Comment by arturbeg** [comment]
> @ioanaroceanu2000 please can you review [af8aec1](https://github.com/Voltz-Protocol/voltz-core/commit/af8aec11b608ee09e47264fa5cb657ebad80e457) carefully because it changes some of the work you did for the Periphery and I'd like you to check I haven't made things worse rather than better. :)

hey @ioanaroceanu2000 can you pls take a look at this when time

**Comment by CyclopsRex** [comment]
Rebased on main. @ioanaroceanu2000 can you please review https://github.com/Voltz-Protocol/voltz-core/pull/197/commits/3443f2a38d7497db1611ec87929f287ed77d7a25 ASAP?

File: deployConfig/poolConfig.ts [hunk 1]
```diff
@@ -53,44 +53,12 @@ export interface poolConfig {
 }
 
 export const poolConfigs: { [name: string]: poolConfig } = {
-  borrow_aETH_v1: {
```
Comment (0xZenus) [inline deployConfig/poolConfig.ts:56]: This is still active right?

File: tasks/irsInstances.ts [hunk 2]
```diff
@@ -70,7 +75,66 @@ async function writeIrsCreationTransactionsToGnosisSafeTemplate(
...
+        }
+      }
+    }
+    builtMapOfRateOracles = true;
+  }
+
+  if (mapOfRateOracleAddressesToNames.has(address.toLowerCase())) {
+    return mapOfRateOracleAddressesToNames.get(address.toLowerCase());
+  } else {
+    return "<unknown>";
```
Comment (0xZenus) [inline tasks/irsInstances.ts:136]: Is there any advantage or returning this instead of throwing error?

File: tasks/upgrades.ts [hunk 3]
```diff
@@ -189,20 +189,21 @@ task(
       }
 
       if (contractName === "Periphery") {
-        // deployedArtifact -> change to Periphery_Implemenatation.sol
-        deployedArtifactImpl = path.join(
+        // Move the new implementation artifact from Periphery.json to Periphery_Implemenatation.json
+        const proxyArtifactPath = path.join(
```
Comment (ioanaroceanu2000) [inline tasks/upgrades.ts:193]: we're making a backup to the Periphery.json contract that contains the proxy logic and the periphery impl logic. If the ABI changes for the periphery, this means the Periphery.json changes too. I think this mistake can be noticed rn, checkout the swap function in Periphery_Implementation (uint margin delta) vs Periphery (int margin delta).
The question is, do we use this 'combined' ABI anywhere? If not, we better just remove it completly.

File: tasks/irsInstances.ts (hunk 2 above)
Comment (CyclopsRex) [reply tasks/irsInstances.ts:136]: Yeah. It's not really an error cos we have some (abandoned) IRS instances that don't use any of the latest rate oracles.

File: deployConfig/poolConfig.ts (hunk 1 above)
Comment (CyclopsRex) [reply deployConfig/poolConfig.ts:56]: Still active to end of year, but not the latest config. Do you think we should be keeping the config for still-active-but-not-the-most-up-to-date IRS instances? Happy to keep if we think we will use it but I wasn't sure what we would use it for.

Comment (0xZenus) [reply deployConfig/poolConfig.ts:56]: yes, I'd keep this until the end of year and we can delete then. Just for consistency.

File: tasks/irsInstances.ts (hunk 2 above)
Comment (0xZenus) [reply tasks/irsInstances.ts:136]: Fair enough.

**Review by ioanaroceanu2000 — APPROVED** [review]

**Pull request merged**